app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,33 @@
+class RecipesController < ApplicationController
+
+  def search; end
+
+  def index
+    terms = search_params["terms"]
+    list = MuncherApiWrapper.list_recipes(terms)
+    @recipe_list = list.paginate(:page => params[:page], :per_page => 8)
+  end
+
+  def show
+    id = recipe_id
+    recipe = MuncherApiWrapper.show_recipe(id)
+
+    if recipe.nil?
+      flash[:notice] = "Something went wrong. Please enter a new search."
```

Comment:
This is where using an exception to report errors comes in handy - an exception contains a lot more useful info than a `nil`.